src/Common/src/Interop/Windows/mincore/Interop.GetConsoleTitle.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Console_L1, CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "GetConsoleTitleW")]
+        private static extern int GetConsoleTitle(StringBuilder title, int nSize);
```

Comment:
> FWIW, I'm not sure how that's relevant in the case of a P/Invoke

On Desktop, at least we used StringBuilder mostly when we needed buffer re-sizing in subsequent calls to the same method and I always looked at it more as a managed layer principle than an interop one. I could be wrong one, but I generally saw this pattern.

> if the P/Invoke were defined with a char* for example, then the initial space could be stack allocated,

The desktop code tries to do something similar not for perf reasons but mainly because it is implemented as a QCall and that was the convention followed for QCalls. As you mentioned, I did not think it was worth using stackalloc for this. I am not opposed to using char* here, I am just not sure it is worth in this particular case.